src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,66 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { spawn } from 'child_process';
+import * as vscode from 'vscode';
+
+import { RazorLogger } from '../RazorLogger';
+
+export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+
+    constructor(private readonly logger: RazorLogger, private readonly vscodeType: typeof vscode) { }
+
+    public async resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration): Promise<vscode.DebugConfiguration | undefined> {
+        const output = this.vscodeType.window.createOutputChannel('Blazor WebAssembly App');
+        const app = spawn('dotnet', ['run'], {
+            cwd: configuration.cwd || folder && folder.uri && folder.uri.fsPath,
+            env: {
+                ...process.env,
+                ASPNETCORE_ENVIRONMENT: 'Development',
+                ...configuration.env,
+            },
+        });
+
+        app.stdout.on('data', (data) => output.append(data.toString()));
+        app.stderr.on('data', (data) => output.append(data.toString()));
+        app.on('error', (error) => {
+            output.append(error.toString());
+            this.logger.logError('[DEBUGGER] Error when launch app: ', error);
+        });
+
+        output.show();
+
+        const browser = {
+            name: '.NET Core Debug Blazor Web Assembly in Browser',
+            type: configuration.browser === 'edge' ? 'pwa-msedge' : 'pwa-chrome',
+            request: 'launch',
+            timeout: configuration.timeout || 30000,
+            url: configuration.url || 'https://localhost:5001',
+            webRoot: configuration.webRoot || '${workspaceFolder}',
+            inspectUri: '{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}',
+            trace: configuration.trace || false,
+        };
+
+        let showErrorInfo = false;
+
+        try {

```

Comment:
Don't you need to wait until the web server is ready before you start the browser? Ex: The web server spits out stdout text of "Now listening on..."

Reply:
The browser will launch and wait for the dev server to launch before timing out. We can wait for the web server to launch before starting the browser but I think the immediate visual feedback of the browser opening up when the user launches the debugger is helpful here.